solarforecastarbiter/metrics/preprocessing.py

```diff
@@ -58,6 +62,75 @@ def apply_validation(obs_df, qfilter, handle_func):
         return validated_obs, counts
 
 
+def apply_fill(fx_data, forecast, forecast_fill_method, start, end):
+    """
+    Apply fill procedure to the data from the start to end timestamps.
+
+    Parameters
+    ----------
+    fx_data : pandas.Series or pandas.DataFrame
+        Forecast data with pandas.DatetimeIndex.
+    forecast : datamodel.Forecast
+    forecast_fill_method : {'drop', 'forward', float}
+        Indicates what process to use for handling missing forecasts.
+          * _'drop'_ drops all missing values for any row with a missing value.
+          * _'forward'_ fills missing values with the most recent real value.
+            If any leading missing values fill with zeros.
+          * _float_ fills any missing values with the given value.
+    start : pandas.Timestamp
+    end : pandas.Timestamp
+
+    Returns
+    -------
+    filled: pandas.Series or pandas.DataFrame
+        Forecast filled according to the specified logic
+    count : int
+        Number of values filled or dropped
+    """
+    forecast_fill_method = str(forecast_fill_method)
+    # Create full datetime range at resolution
+    full_dt_index = pd.date_range(
+        start=start, end=end, freq=forecast.interval_length,
+        closed=datamodel.CLOSED_MAPPING[forecast.interval_label],
+        name=fx_data.index.name)
+    try:
```

Comment:
sorry if I was unclear in my previous comment.

This is too much code in a `try/except`. I was looking for something like:

```python
if forecast_fill_method == 'drop':
    # drop implementation
elif forecast_fill_method == 'forward':
    # forward implemention
else:
    # we'll have a problem if this isn't a float
    try:
        const_fill_value = pd.to_numeric(forecast_fill_method).astype(float)
    except ValueError:
        # apparently it was not a float, so raise more descriptive message
        raise ValueError('Unsupported...')
    # it was float-ish so now we can perform the reindex and fill
    # fill with float implementation
```

Reply:
Oh yeah that's much better and isolates it

Reply:
Oh. Also do you want `bool` to be supported e.g., `'True'`. Based on what I saw for events and the dtype forced to be `float` I figure it makes more sense just to to use `'1'` and `'0'`. But that's any easy change although I'll have to add a test for them.

Reply:
If I understand correctly, the event forecasts are still floats when this is called. Then they're casted into bools a few lines below in `resample_and_align`. So it's ok with me to require 1 and 0 here.